src/main/java/com/aliction/git/remote/integration/GitHubIntegration.java

```diff
@@ -0,0 +1,93 @@
+package com.aliction.git.remote.integration;
+
+import java.io.IOException;
+
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+import com.aliction.git.properties.GitRemoteProperties;
+
+public class GitHubIntegration implements GitRemoteIntegration {
+
+    GitHub github;
+    String user;
+    GHRepository repo = null;
+    GitRemoteProperties props;
+    Boolean usingToken;
+    CredentialsProvider credentialsProvider;
+    String remoteURL = null;
+
+    public GitHubIntegration() {
+        // TODO Auto-generated constructor stub
+    }
+
+    @SuppressWarnings("deprecation")
+    public GitHubIntegration(GitRemoteProperties properties) {
+        // TODO Auto-generated constructor stub
+        props = properties;
+        try {
+            if (props.getToken().isEmpty()) {
+                usingToken = false;
+                System.out.println("Unsecured connection using username and password since token is not provided in properties file");
+                github = GitHub.connectToEnterprise(props.getRemoteGitUrl(), props.getLogin(), props.getPassword());
```

Comment:
enterprise only? we may need to add more flags to support enterprise and public

Reply:
@porcelli, it is not enterprise only, the enterprise method can work smoothly if you provided the public URL"api.github.com", so it works for both.
This will work better, instead of me having two different methods one for public and the other for enterprise.